crates/dns-resolver/src/lib.rs

```diff
@@ -17,8 +17,8 @@ mod resolver;
 pub use resolver::UnboundResolver;
 pub use resolver::{HickoryResolver, Resolver};
 
-static RESOLVER: LazyLock<ArcSwap<Resolver>> =
-    LazyLock::new(|| ArcSwap::from_pointee(default_resolver()));
+static RESOLVER: LazyLock<ArcSwap<Box<dyn Resolver>>> =
```

Comment:
ah, makes sense; please add a comment here in the code about that, because it looks redundant and it was the first thing my brain wanted to comment on!

I agree, the double indirection should be tiny in cost given that we should only go through this on a cache miss and will need to go out to the network.

Reply:
Will add a comment.